### PR TITLE
[DependencyInjection] update service_container.rst

### PR DESCRIPTION
**Pull request by mohamedGasmii**
Add alternative solution.

### CODE REVIEW

File: service_container.rst [hunk 1]
```diff
@@ -1194,6 +1194,110 @@ and ``site_update_manager.normal_users``. Thanks to the alias, if you type-hint
...
+                    - '@mailer'
+                    - 'contact@example.com'
+
+            # Create an alias, so that - by default - if you type-hint SiteUpdateManager,
+            # the site_update_manager.superadmin will be used
+            App\Service\SiteUpdateManager: '@site_update_manager.superadmin'
+
+            # If you type-hint SiteUpdateManager
+            # for a ``$normalUsers`` argument is detected.
+            App\Service\SiteUpdateManager $normalUsers: '@site_update_manager.normal_users'
```
Comment (OskarStark) [inline service_container.rst:1231]: @nicolas-grekas can you confirm? TBH I never heard/saw such thing.

Should we add it to the docs?

Thanks

File: service_container.rst [hunk 2]
```diff
@@ -1194,6 +1194,110 @@ and ``site_update_manager.normal_users``. Thanks to the alias, if you type-hint
 ``SiteUpdateManager`` the first (``site_update_manager.superadmin``) will be passed.
 If you want to pass the second, you'll need to :ref:`manually wire the service <services-wire-specific-service>`.
 
+Another solution is to add another id ``site_update_manager.superadmin`` followed by
+a variable name matching the one you use when doing the injection::
```
Comment (nicolas-grekas) [inline service_container.rst:1198]:
```suggestion

If you want to pass ``site_update_manager.superadmin``, you'll need either to :ref:`manually wire it<services-wire-specific-service>`
or to create a named autowiring alias.
```

(named autowiring alias should link to service_container/autowiring.rst, then, the example is not needed)

File: service_container.rst (hunk 1 above)
Comment (nicolas-grekas) [reply service_container.rst:1231]: yes, it's a named autowiring alias, and it's good to tell about it indeed
that being said, we have the `service_container/autowiring.rst` that already explains what they are.
I think the example should be removed and replaced by a link to that page.

**Comment by OskarStark** [comment]
Closed via https://github.com/symfony/symfony-docs/pull/18749

**Pull request closed without merging**